docs/planning/sprintplanning.md

```diff
@@ -0,0 +1,66 @@
+- priority/p0 is reserved for critical bugs that needs to be addressed immediately in current release at the cost of currently planned items.
+- WG lead will drive this and his/her decision will be final
+
+## How to manage Story/Feature?
+- Story/Feature typically spans 1-2 weeks for one engineer.
+To propose a new feature,  create a new issue of type “Feature Request”
+- Please fill in as much detail as possible in the Feature template and propose a release cycle
+- All new features will be labelled “kind/feature”, and “triage/needstriage”
+- All new proposals are discussed either in WG meeting, or on slack, or dedicated meetings if needed at any point scheduled by the proposer. 
+- By R-2w’s WG meeting all new proposals should be vetted and verified that they fit the [Planning criteria](#planning-criteria). Features that do not fit this criteria will not be considered in the current release planning cycle and hence early discussion is encouraged instead of waiting for R-2w.
+- All eligible proposals will then be snapshotted into the voting document and open for voting till R-1w WG meeting. In R-1w WG meeting top features based on votes will be planned for next release. 
+> We need some kind of estimates for each feature for release planning. Shall we adopt story points on fibonacci scale or t-shirt sizing? My vote goes to story points on fibonacci scale to avoid spending a lot of time on estimation, which is a common pitfall.
+- Selected features will be labeled “triage/accepted”. Rest will be labelled “triage/ice-boxed” and will be eligible for next planning cycle. Accepted features will be added to the correct milestone and sprint specific project for tracking.
+
+## How to manage Epic/Scenario?
+- Epic/Scenario typically spans 1-3 months or even more in some cases.
+- For new proposals these will be created and triaged/refined just like Story/Feature. However, after triaging these need to be broken down into Features for planning and voting.
+- Github doesn’t allow nested Issues, hence the only good way to track these are by creating Projects and adding the broken down features as issues to this project. An issue can be part of multiple projects and hence we can get a Scenario\Epic view as well as a Sprint view through respective projects.
+
+## Planning criteria:
+- Features are well defined and scoped with clear exit criteria. No feature creeps once release starts
+- Features can be implemented in <2w by one engineer. In case of Epic/Scenario, it could be broken down into <2w features with clear exit criteria.
+- The engineering team should be confident that any engineer can implement the feature based on the content in the feature *(This is similar to refinement in a scrum, but doesn’t have to be as strict as a functional specification)*.
+
+## Organizing with Github Projects 
+- New Project per sprint. Today, the sprint and milestone are both aligned. However in future we may decide to have multiple sprints in a release cycle.
```

Comment:
I'm loath to have both milestones and projects to track the same thing. "A man with a watch knows what time it is. A man with two watches is never sure."

Reply:
haha. These are similar frustrations when I started exploring Github for project management :D
See this: https://github.com/knative/eventing/pull/910#discussion_r266630757 

Question: Do we need a board with columns such as Todo, InProgress, ReviewInProgress ReviewerApproved, Done? In teams I have worked previously I have found this to be super useful for the team to prioritize and complete (Done) as many committed features as possible rather than  end sprint with most features "InProgress"

A good thing I liked about GitHub is that with AutoKanban boards no one has to manage the board, it updates itself as the Issues/PRs update themselves. 

If we don't need it then Milestone is good enough. If we need this then lets create a project and try out how it works. It too taxing, then dump it, else continue using it.

Reply:
I'm fine with trying and seeing what works. We can use GitHub's stock project automation, or add automation with Prow (when https://github.com/kubernetes/test-infra/pull/10401 eventually gets merged)

Reply:
Yup. We can start with AutomatedKanWithReviews and if we think it is helpful then can improve it with Prow. Or else discard the idea.
One thing we should definitely avoid is another tax to maintain this board. Automating it with Prow based on labels will be ideal.

Reply:
## TL;DR
Given there's no strong buy-in and no clear gains over the overhead, we start without creating a project per sprint and once we start having pain points that make this a clear win, we start using it. Details below.

## Details
I keep getting mixed feelings about creating a board and a milestone for the Sprint. I've worked a lot with all sort of agile boards tools (scrum, kanban, Jira, ZenHub etc) and usually I do like to have a single board tracking the current iteration/sprint whatever you wanna name it. However the given approach of mixed use of projects to track both "goals/epics" and "sprint" has following pros & cons IMHO

### Pros
-  Visualization and transparency of the "review" and "approve" stages of milestone scheduled work items (milestones already visualizes the todo - in progress - completed stages)
- Single consolidated view of the state of the milestone that can show bottlenecks (issues are piling in review, or in approve). 

### Cons
- Added overhead of maintaining the board (not sure if it can be solved by github automation?)
- A bit confusing when visiting the "Projects" tab and seeing "0.6 Sprint" and "Broker" projects side by side.

In such cases of no clear win and no strong buy-in from team members, I like to start the other way around. Let's see what are the big pain points that we want to solve, and use the tool to solve that. 

IMHO the biggest pain points that I faced here were:
- Having a clear process from ideas to stories to planned work to execution.
- Understanding what's planned (roadmap, goals/epics..etc) and what is scheduled soon so that I can pick up what to work on next.

Given the engagement level of working group members and their speed, I didn't see a problem in delayed "reviews" or "approvals" and this IMHO makes using a project for a sprint more cons than pros. 

Hence after writing this actually I suggest to start without a "Sprint" project and if we face such issues as bottlenecks in reviews or approvals then we can start using them and the gain will be very visible for everyone.